src/System.Private.ServiceModel/tests/Scenarios/Binding/Custom/CustomBindingTests.cs

```diff
@@ -51,4 +51,69 @@ public static void DefaultSettings_Tcp_Binary_Echo_RoundTrips_String()
             ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
         }
     }
+
+    public class MessageModifier : ChannelMessageInterceptor
+    {
+        int send = 0;
+
+        public override void OnSend(ref Message msg)
+        {
+            send++;
+            if (send % 2 == 0)
+            {
+                // Add extra header so that when the message arrives on service side, the message won't be dropped
+                msg.Headers.Add(MessageHeader.CreateHeader("ByPass", "urn:InterceptorNamespace", "ByPassPassword"));
+            }
+        }
+
+        public override ChannelMessageInterceptor Clone()
+        {
+            return new MessageModifier();
+        }
+    }
+
+    [Fact]
+    [OuterLoop]
+    [ActiveIssue(1170)]
+    public static void CustomBinding_Message_Interceptor()
+    {
+        ChannelFactory<IChannelExtensibilityContract> factory = null;
+        IChannelExtensibilityContract serviceProxy = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            CustomBinding binding = new CustomBinding(
+                new InterceptingBindingElement(new MessageModifier()),
+                new HttpTransportBindingElement());
+
+            factory = new ChannelFactory<IChannelExtensibilityContract>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_ChannelExtensibility));
+            serviceProxy = factory.CreateChannel();
```

Comment:
I think we shouldn't need factory.Open() for it to work. In fact, I've tried to call factory.Open() first but it still fails.
I've put an break at where the inner factory is open and that code has never been executed so I think there is something wrong because class InterceptingChannelFactory<TChannel>
.OnOpen() method is never invoked, whereas my understanding is that it should.